File: src/engine/transpile/src/test/resources/golden/SYK001/WS_エラーメッセージ.py
"""SYK001 のデータ部レコード WS-エラーメッセージ を逐語対訳した自動生成コード。"""
from cobol_runtime import (
    decode_alphanumeric,
    decode_binary,
    decode_packed,
    decode_zoned,
    encode_alphanumeric,
    encode_binary,
    encode_packed,
    encode_zoned,
)


class WS_エラーメッセージ:
    """01 WS-エラーメッセージ 総バイト長 40。"""

    LENGTH = 40

    def __init__(self, data=None):
        if data is None:
            self._data = bytearray(40)
        else:
            self._data = bytearray(data)
            if len(self._data) < 40:
                self._data.extend(b"\x00" * (40 - len(self._data)))

    @property
    def data(self):
        return self._data

    # WS-エラーメッセージ
    def get_WS_エラーメッセージ(self):
        return decode_alphanumeric(self._data, 0, 40)
    def set_WS_エラーメッセージ(self, value):
        encode_alphanumeric(self._data, 0, 40, value)

File: src/engine/transpile/src/test/resources/golden/SYK001/WS_エラーメッセージ.java
package cobolinsight.generated;

import cobolinsight.runtime.CobolRuntime;
import java.util.Arrays;

/** SYK001 のデータ部レコード WS-エラーメッセージ を逐語対訳した自動生成コード。 */
public final class WS_エラーメッセージ {

    /** 01 WS-エラーメッセージ 総バイト長 40。 */
    public static final int LENGTH = 40;

    private final byte[] data;

    public WS_エラーメッセージ() {
        this.data = new byte[40];
    }

    public WS_エラーメッセージ(byte[] source) {
        this.data = Arrays.copyOf(source, 40);
    }

    public byte[] data() {
        return data;
    }

    // WS-エラーメッセージ
    public String get_WS_エラーメッセージ() {
        return CobolRuntime.decodeAlphanumeric(data, 0, 40);
    }
    public void set_WS_エラーメッセージ(String value) {
        CobolRuntime.encodeAlphanumeric(data, 0, 40, value);
    }
}
